docs/source/guide/guide-executors.rst

```diff
@@ -203,6 +203,74 @@ You can also include both noise models and finite sampling in your executor.
     assert noisy_output < 1.0
     assert noisy_output > 0.5
 
+.. _pyquil_executors:
+
+PyQuil Executors
+================
+
+This section contains executors for use with `pyQuil <https://github.com/rigetti/pyquil>`_.
+
+PyQuil: Quantum Cloud Services
+------------------------------
+
+This executor can be used to run on `Quantum Cloud Services <https://arxiv.org/abs/2001.04449>`_
+(QCS), the hardware platform provided by Rigetti Computing. Requires a QCS account and
+reservation on a quantum processor (QPU).
+
+In addition, ``mitiq_pyquil/executors.py`` has a function for generating a
+QCS executor of this form.
+
+.. code-block::
+
+    from pyquil import Program, get_qc
+    from pyquil.gates import MEASURE, RESET
+
+    from mitiq.mitiq_pyquil.compiler import basic_compile
+    from mitiq.mitiq_pyquil.executor import zz_expectation
+
+    # this will only work if running on QCS and engaged to Aspen-8
+    qpu = get_qc("Aspen-8")
+
+    def executor(program: Program, shots: int = 1000) -> float:
+        p = Program()
+
+        # add reset
+        p += RESET()
+
+        # add main body program
+        p += program.copy()
+
+        # add memory declaration
+        qubits = p.get_qubits()
+        ro = p.declare("ro", "BIT", len(qubits))
+
+        # add measurements
+        for idx, q in enumerate(qubits):
+            p += MEASURE(q, ro[idx])
+
+        # add numshots
+        p.wrap_in_numshots_loop(shots)
+
+        # nativize the circuit
+        p = basic_compile(p)
+
+        # compile the circuit
+        b = qpu.compiler.native_quil_to_executable(p)
+
+        # run the circuit, collect bitstrings
+        qpu.reset()
+        results = qpu.run(b)
+
+        # compute ZZ expectation value
+        return zz_expectation(results)
+
+    program = Program()
+    program += X(0)
+    program += X(1)
+
+    # should give 1.0 with a noiseless backend
+    executor(program)
```

Comment:
Since this code is not tested and duplicated from `executor.py`, I propose linking to the function in `executor.py` (API doc or otherwise), or adding a test here.

Reply:
Yeah I wanted to test it but the Aspen-8 thing causes issues obviously. I'll think on it a bit more.